commons-logs/src/main/java/eu/europeana/api/commons/logs/LoggableDispatcherServlet.java

```diff
@@ -0,0 +1,213 @@
+        logMessage.setPort(requestToCache.getServerPort());
+        logMessage.setJavaMethod(handler.getHandler().toString());
+
+        // get the client's Ip and Geo Location
+        String clientIP = getClientIP(requestToCache);
+        // TODO Verify which Ip is this exactly; for now getting IPV4
+        String ipV4 = getIPV4();
+
+        logMessage.setClientIPv4(clientIP);
+        logMessage.setIpV4(ipV4);
+        logMessage.setClientLocation(getGeoLocation(clientIP));
+        logMessage.setLocation(getGeoLocation(ipV4));
+
+        // TODO figure out how we will calculate this
+        // Gorouter Time : is the total time it takes for the request to go through the
+        // Gorouter initially plus the time it takes for the response to travel back through the Gorouter. This does not include
+        // the time the request spends traversing the network to the app. This also does not include the time the
+        // app spends forming a response.
+        //logMessage.setGorouterTime(goRouterTime);
+
+        logMessage.setHttpVersion(StringUtils.substringAfter(requestToCache.getProtocol(), "/"));
+        logMessage.setReferer(requestToCache.getHeader(LogConstants.REFERER));
+        logMessage.setHttpStatus(responseToCache.getStatus());
+        logMessage.setProcessTime(processTime);
+        // server date and time
+        logMessage.setServerDate(StringUtils.substringBefore(serverTime, "T"));
+        logMessage.setServerTime(StringUtils.substringBetween(serverTime, "T", "Z"));
+        logMessage.setServerTimeZoneOffset(LogConstants.SERVER_TIMEZONE_OFFSET);
+
+        logMessage.setUrlQuery(requestToCache.getQueryString());
+        logMessage.setUrlPath(requestToCache.getPathInfo());
+        logMessage.setUserAgent(requestToCache.getHeader(LogConstants.USER_AGENT));
+        logMessage.setVcapRequestId(requestToCache.getHeader(LogConstants.X_VCAP_REQUEST_ID));
+        logMessage.setMethod(requestToCache.getMethod());
+        logMessage.setxB3parentSpanId(requestToCache.getHeader(LogConstants.X_B3_PARENT_SPAN_ID));
+        logMessage.setxB3SpanId(requestToCache.getHeader(LogConstants.X_B3_SPAN_ID));
+        logMessage.setxB3TraceId(requestToCache.getHeader(LogConstants.X_B3_TRACE_ID));
+        logMessage.setxGlobalTransId(requestToCache.getHeader(LogConstants.X_GLOBAL_TRANSACTION_ID));
+        logMessage.setCfConnectingIp(requestToCache.getHeader(LogConstants.CF_CONNECTING_IP));
+        logMessage.setCfIpCountry(requestToCache.getHeader(LogConstants.CF_IP_COUNTRY));
+
+        LOG.info(logMessage);
+    }
+
+    /**
+     * Gets the current date and time in yyyy-MM-dd'T'HH:mm:ss.SSS'Z' format
+     *
+     * @return date in yyyy-MM-dd'T'HH:mm:ss.SSS'Z' format
+     */
+    private String getCurrentDateTime() {
+        DateFormat df = new SimpleDateFormat(LogConstants.DATE_FORMAT, Locale.US);
+        df.setTimeZone(TimeZone.getTimeZone(LogConstants.TIME_ZONE));
+        return df.format(new Date());
```

Comment:
This look expensive. Don't we have a date in the incoming request we can copy?

Reply:
Unfortunately, we don't have any date in the request header coming. That's why I added it